### PR TITLE
allow to store meta information about Page Object fields

### PR DESCRIPTION
**Pull request by kmike**
Motivation:

1. Allow custom extensions in third-party code, a point where additional information about a field can be stored
2. Introduce a structured FieldInfo, which can be used in other features

### CODE REVIEW

File: web_poet/fields.py [hunk 1]
```diff
@@ -25,22 +25,34 @@ async def to_item(self):
...
     Page Object method decorated with ``@field`` decorator becomes a property,
     which is used by :func:`item_from_fields` or :func:`item_from_fields_sync`
     to populate item attributes.
 
     By default, the value is computed on each property access.
     Use ``@field(cached=True)`` to cache the property value.
+
+    ``meta`` parameter allows to store arbitrary information for the
+    field - e.g. ``@field(meta={"expensive": True})``. This information
+    can be later retrieved for all fields using :func:`fields_dict` function.
```
Comment (Gallaecio) [inline web_poet/fields.py:55]:
```suggestion
    The ``meta`` parameter allows to store arbitrary information for the field, 
    e.g. ``@field(meta={"expensive": True})``. This information can be later 
    retrieved for all fields using the :func:`fields_dict` function.
```

**Review by Gallaecio — APPROVED** [review]

**Review by BurnzZ — COMMENTED** [review]
Hi @kmike , I think I'm missing some context regarding the applications of `fields_dict`. Perhaps we should create a section in the docs for it?

**Comment by kmike** [comment]
> Hi @kmike , I think I'm missing some context regarding the applications of fields_dict. Perhaps we should create a section in the docs for it?

Makes sense! I've added some docs; could you please check it @BurnzZ?

File: docs/advanced/fields.rst [hunk 2]
```diff
@@ -489,3 +489,31 @@ with async versions of ``@property`` and ``@cached_property`` decorators; unlike
...
+returns a dictionary, where keys are field names, and values are
+:class:`web_poet.fields.FieldInfo` instances.
+
+.. code-block:: python
+
+    from web_poet.fields import get_fields_dict
+
+    fields_dict = get_fields_dict(MyPage)
+    field_names = fields_dict.keys()
+    my_field_meta = fields_dict["my_field"].meta
```
Comment (BurnzZ) [inline docs/advanced/fields.rst:519]:
```suggestion
    my_field_meta = fields_dict["my_field"].meta
    
    print(field_names )  # dict_keys(['my_field'])
    print(my_field_meta) # {'expensive': True}
```

File: docs/advanced/fields.rst [hunk 3]
```diff
@@ -489,3 +489,31 @@ with async versions of ``@property`` and ``@cached_property`` decorators; unlike
 :func:`functools.lru_cache`, they all work fine for this use case.
 
 .. _async_property: https://github.com/ryananguiano/async_property
+
+Field metadata
+--------------
+
+web-poet allows to store arbitraty information for each field, using
```
Comment (BurnzZ) [inline docs/advanced/fields.rst:496]:
```suggestion
``web-poet`` allows to store arbitrary information for each field, using
```

**Review by BurnzZ — APPROVED** [review]
LGTM @kmike ! 🙌

**Pull request merged**